src/typescript/Scala.tmLanguage.ts

```diff
@@ -677,21 +676,55 @@ export const scalaTmLanguage: TmLanguage = {
             }
           }
         },
-        {
-          match: '(==?|!=|<=|>=|<>|<|>)',
-          name: 'keyword.operator.comparison.scala'
-        },
-        {
-          match: '(\\-|\\+|\\*|/(?![/*])|%|~)',
-          name: 'keyword.operator.arithmetic.scala'
+        { // Operators with three or more characters
+          match: `(${opchar}|[\\\\]){3,}`,
+          name: 'keyword.operator.scala'
         },
-        {
-          match: `(?<!${opchar}|_)(!|&&|\\|\\|)(?!${opchar})`,
-          name: 'keyword.operator.logical.scala'
+        { // Operators with two characters
+          match: `((?:${opchar}|[\\\\]){2,}|_\\*)`,
+          captures: {
+            '1': {
+              patterns: [
+                {
+                  match: '(\\|\\||&&)',
+                  name: 'keyword.operator.logical.scala'
+                },
+                {
+                  match: '(\\!=|==|\\<=|>=)',
+                  name: 'keyword.operator.comparison.scala'
+                },
+                {
+                  match: '..',
+                  name: 'keyword.operator.scala'
+                }
+              ]
+            }
+          }
         },
-        {
-          match: '(<-|←|->|→|=>|⇒|\\?|\\:+|@|\\|)+',
-          name: 'keyword.operator.scala'
+        { // Operators with one character
+          match: `(?<!_)(${opchar}|\\\\)`,
+          captures: {
+            '1': {
+              patterns: [
+                {
+                  match: '(\\!)',
+                  name: 'keyword.operator.logical.scala'
+                },
+                {
+                  match: '(\\*|-|\\+|/|%|~)',
+                  name: 'keyword.operator.arithmetic.scala'
+                },
+                {
+                  match: '(=|\\<|>)',
```

Comment:
Here I think we shouln't match `=`, as a single `=` doesn't compare but defines things.